packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Writers/CodeWriter.cs

```diff
@@ -386,10 +386,10 @@ public void WriteParameter(ParameterProvider parameter)
             AppendRawIf("ref ", parameter.IsRef);
 
             Append($"{parameter.Type} {parameter.Name:D}");
-            if (parameter.DefaultValue != null)
+            if (parameter.InitializationValue != null)
             {
                 AppendRaw(" = ");
-                parameter.DefaultValue.Write(this);
+                parameter.InitializationValue.Write(this);
```

Comment:
I am not sure this is correct.  Here we are writing the signature which should use the default value not the initialization value?

Reply:
I kind of think they should be the same thing for a parameter.

Reply:
I've reverted this to `DefaultValue.` The reason I've added `InitializationValue` is because I think there is a use case where we need to provide a parameter's value when creating the constructor overload(s) for a client but not necessarily assign it in the base parameter signature.. ie:

```c#
 public OpenAIClient(ApiKeyCredential credential) : this(new Uri("https://api.openai.com/v1"), credential, new OpenAIClientOptions())
 { }

 public OpenAIClient(Uri endpoint, ApiKeyCredential credential, OpenAIClientOptions options)
 { }
```

If we only used `DefaultValue` in it's current implementation this is what the generated code would look like instead:
```c#
 public OpenAIClient(ApiKeyCredential credential) : this(new Uri("https://api.openai.com/v1"), credential, new OpenAIClientOptions())
 { }

 public OpenAIClient(Uri endpoint = new Uri("https://api.openai.com/v1"), ApiKeyCredential credential, OpenAIClientOptions options = new OpenAIClientOptions())
 { }
```

Do let me know if we have suggestions on how we can improve this if what I'm proposing is not the best solution.

Reply:
I am not saying there isn't such a thing as an Initialization value but it is different from default value.

for instance I can do this

```c#
public void Foo(int? x = null) : base(x ?? GetSomeValue())
```

The expression for default is `Null` the expression for intialize is `NullCoalesce(x, Static.Invoke("GetSomeValue"))`.  They are not the same.

Reply:
OK in this perspective they are different, I see it now